src/num2words.rs

```diff
@@ -319,13 +319,11 @@ mod tests {
 
     #[test]
     fn test_string_not_valid() {
-        match Num2Words::parse("not a number") {
-            Some(_) => assert!(false),
-            None => assert!(true),
+        if Num2Words::parse("not a number").is_some() {
```

Comment:
LGTM on most side here, I am just not a huge fan of the idea of using `panic` which will 1. stop the unit tests here, without running the other ones and 2. don't report anything if it works.

The following should do a better job:
```rust
assert!(Num2Words::parse("not a number").is_none());
assert!(Num2Words::parse("NAN").is_none());
```

Reply:
I just fixed https://rust-lang.github.io/rust-clippy/rust-1.74.0/index.html#/assertions_on_constants
I agree your version is better.

Reply:
Related comment for tracking: https://github.com/Ballasi/num2words/commit/a109dbcb0914dc897f17d9b709a30206bf1d9bd5#r133143809